src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -512,13 +509,39 @@ TypeSymbol BindRecursivePatternType(TypeSyntax typeSyntax, TypeSymbol inputType,
             }
         }
 
-        private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node, TypeSymbol inputType, bool hasErrors, DiagnosticBag diagnostics)
+        private BoundPattern BindRecursivePattern(RecursivePatternSyntax node, TypeSymbol inputType, bool hasErrors, DiagnosticBag diagnostics)
         {
             TypeSyntax typeSyntax = node.Type;
             TypeSymbol declType = BindRecursivePatternType(typeSyntax, inputType, diagnostics, ref hasErrors, out BoundTypeExpression boundDeclType);
 
-            var patterns = ArrayBuilder<BoundSubpattern>.GetInstance(node.SubPatterns.Count);
             MethodSymbol deconstructMethod = null;
+            ImmutableArray<BoundSubpattern> deconstructionSubpatterns = default;
+            if (node.DeconstructionSubpattern != null)
+            {
+                deconstructionSubpatterns = BindDeconstructionSubpattern(node.DeconstructionSubpattern, declType, diagnostics, out deconstructMethod, ref hasErrors);
+            }
+
+            ImmutableArray<BoundSubpattern> propertiesOpt = default;
+            if (node.PropertySubpattern != null)
+            {
+                propertiesOpt = BindPropertySubpattern(node.PropertySubpattern, declType, diagnostics, ref hasErrors);
+            }
+
+            BindPatternDesignation(node, node.Designation, declType, typeSyntax, diagnostics, ref hasErrors, out Symbol variableSymbol, out BoundExpression variableAccess);
+            return new BoundRecursivePattern(
+                syntax: node, declaredType: boundDeclType, inputType: inputType, deconstructMethodOpt: deconstructMethod,
+                deconstruction: deconstructionSubpatterns, propertiesOpt: propertiesOpt, variable: variableSymbol, variableAccess: variableAccess, hasErrors: hasErrors);
```

Comment:
>deconstruction [](start = 16, length = 14)

Should `deconstruction:` be `deconstructionOpt:`?

Reply:
Probably, but should *all* the properties be Opt?

---
In reply to: [187495562](https://github.com/dotnet/roslyn/pull/26782#discussion_r187495562) [](ancestors = 187495562)

Reply:
It struck me as odd that we have `deconstruction:` and `propertiesOpt:`. Maybe remove `Opt` then?